core/network/fabric-interop-cc/contracts/interop/manage_assets.go

```diff
@@ -30,15 +32,57 @@ type AssetLockValue struct {
 	ExpiryTimeSecs	uint64	`json:"expiryTimeSecs"`
 }
 
-// function to generate contract id as "SHA256" hash based on the asset lock key (which is combination of asset type and asset it)
-func generateAssetLockKeyAndContractId(assetAgreement *common.AssetExchangeAgreement) (string, string) {
+// Object used in the map, contractId --> <asset-type, num-units, locker, ...> (for fungible assets)
+type FungibleAssetLockValue struct {
+	Type		string	`json:"type"`
+	NumUnits	uint64	`json:"numUnits"`
+	Locker		string	`json:"locker"`
+	Recipient	string	`json:"recipient"`
+	Hash		string	`json:"hash"`
+	ExpiryTimeSecs	uint64	`json:"expiryTimeSecs"`
+}
+
+const(
+	assetKeyPrefix		= "AssetKey_"	// prefix for the map, asset-key --> asset-object
+	assetKeyDelimiter	= "_"		// delimiter for the asset-key
+	contractIdPrefix	= "ContractId_"	// prefix for the map, contractId --> asset-key
+)
+
+// function to generate a "SHA256" hash in base64 format for a given the preimage
+func generateSHA256HashInBase64Form(preimage string) string {
 	hasher := sha256.New()
-	assetLockKey := assetAgreement.Type + assetAgreement.Id
-	hasher.Write([]byte(assetLockKey))
+	hasher.Write([]byte(preimage))
 	shaHash := hasher.Sum(nil)
-	shaBase64 := base64.StdEncoding.EncodeToString(shaHash)
-	return assetLockKey, shaBase64
+	shaHashBase64 := base64.StdEncoding.EncodeToString(shaHash)
+	return shaHashBase64
 }
+
+/*
+ * Function to generate asset-lock key (which is combination of asset-type and asset-id)
+ * and contract-id (which is a hash on asset-lock key) for the non-fungible asset locking on the ledger
+ */
+func generateAssetLockKeyAndContractId(assetAgreement *common.AssetExchangeAgreement) (string, string) {
+	assetLockKey := assetKeyPrefix + assetAgreement.Type + assetKeyDelimiter + assetAgreement.Id
```

Comment:
Upon further reflection, you can use the composite key creation function here rather than explicitly specify prefixes and delimiters. This will also allow us to query on the basis of just asset type, should the need arise.

Reply:
I assume that you are suggesting something like this https://stackoverflow.com/questions/52348514/how-to-make-composite-key-for-a-hash-map-in-golang to be used here too.
I want to figure this out later and include the code changes appropriately along with other code changes that I will be doing to address the query functions. Didn't do any changes now. Hope this is fine with you too @VRamakrishna .

Reply:
@VRamakrishna , created composite function using the asset-type and asset-id attributes as suggested (https://pkg.go.dev/github.com/hyperledger/fabric-chaincode-go/shim#ChaincodeStub.CreateCompositeKey).